python-backend/app/api/nris_services/NRIS_service.py

```diff
@@ -4,45 +4,57 @@
 import json
 import functools
 
+from dateutil.relativedelta import relativedelta
 from datetime import datetime
 from flask import request, current_app
 from flask_restplus import Resource, reqparse
 from werkzeug.datastructures import FileStorage
 from werkzeug import exceptions
 from sqlalchemy.exc import DBAPIError
+from app.extensions import cache
+from ..constants import NRIS_CACHE_PREFIX, TIMEOUT_24_HOURS, TIMEOUT_12_HOURS
 
 
-def get_NRIS_token():
-    params = {
-        'disableDeveloperFilter': 'true',
-        'grant_type': 'client_credentials',
-        'scope': 'NRISWS.*'
-    }
-    url = current_app.config['NRIS_TOKEN_URL']
-    if url is None:
-        raise TypeError('Could not load the NRIS URL.')
-    else:
-        resp = requests.get(
-            url=url,
-            params=params,
-            auth=(current_app.config['NRIS_USER_NAME'], current_app.config['NRIS_PASS']))
-        try:
-            resp.raise_for_status()
-        except:
-            raise
+def _get_datetime_from_NRIS_data(date):
+    return datetime.strptime(date, '%Y-%m-%d %H:%M')
+
+
+def _get_NRIS_token():
+    result = cache.get(NRIS_CACHE_PREFIX + 'token')
 
-        if resp.status_code != 200:
-            return None
+    if result is None:
+        params = {
+            'disableDeveloperFilter': 'true',
+            'grant_type': 'client_credentials',
+            'scope': 'NRISWS.*'
+        }
+        url = current_app.config['NRIS_TOKEN_URL']
+        if url is None:
+            raise TypeError('Could not load the NRIS URL.')
+        else:
+            resp = requests.get(
+                url=url,
+                params=params,
+                auth=(current_app.config['NRIS_USER_NAME'], current_app.config['NRIS_PASS']))
+            try:
```

Comment:
Instead of raising the exception and checking the response like this. Can you not check the status code and then return appropriate response?
Is NRIS doing someting funky like returning 200 at HTTP level and then 400 in the body because of which this is required?

Reply:
This call is nested in another api call so I need to raise the exception up a level so that the top level request can return the correct response.